File: data/readable_format/fortran/file943.f90
program main
  implicit none
  integer, parameter :: n = 32
  real :: a(n), b(n)
  integer :: i
  logical :: test_passed

  ! Initialize arrays
  do i = 1, n
    a(i) = 1.0
    b(i) = 0.0
  end do

  ! First modification and test
  do i = 1, n
    b(i) = 2.0
    a(i) = a(i) + b(i)
  end do

  ! Test the results
  test_passed = .true.
  do i = 1, n
    if (a(i) /= 3.0 .or. b(i) /= 2.0) then
      test_passed = .false.
      exit
    endif
  end do

  if (test_passed) then
    print *, 'Test 1 passed: a(i) is 3.0 and b(i) is 2.0 for all i.'
  else
    print *, 'Test 1 failed.'
  endif

  ! Second modification and test
  do i = 1, n
    b(i) = 3.0
    a(i) = a(i) + b(i)
  end do

  ! Test the results
  test_passed = .true.
  do i = 1, n
    if (a(i) /= 6.0 .or. b(i) /= 3.0) then
      test_passed = .false.
      exit
    endif
  end do

  if (test_passed) then
    print *, 'Test 2 passed: a(i) is 6.0 and b(i) is 3.0 for all i.'
  else
    print *, 'Test 2 failed.'
  endif

end program main

File: data/readable_format/cpp/file943.cpp
#include <iostream>
#include <vector>

// Function declarations
void initializeAndModifyArrays(std::vector<float> &a, std::vector<float> &b);
void testArrayValues(const std::vector<float> &a, const std::vector<float> &b, float expectedA, float expectedB, const std::string &testName);

int main() {
    const int n = 32;
    std::vector<float> a(n), b(n);

    // Initialize and modify arrays as per the original program logic
    initializeAndModifyArrays(a, b);

    // Run tests
    testArrayValues(a, b, 6.0f, 3.0f, "Final State Test");

    return 0;
}

void initializeAndModifyArrays(std::vector<float> &a, std::vector<float> &b) {
    const int n = a.size();

    // Initialize arrays
    for (int i = 0; i < n; ++i) {
        a[i] = 1.0f;
        b[i] = 0.0f;
    }

    // First modification
    for (int i = 0; i < n; ++i) {
        b[i] = 2.0f;
        a[i] = a[i] + b[i];
    }

    // Test after first modification
    testArrayValues(a, b, 3.0f, 2.0f, "First Modification Test");

    // Second modification
    for (int i = 0; i < n; ++i) {
        b[i] = 3.0f;
        a[i] = a[i] + b[i];
    }
}

void testArrayValues(const std::vector<float> &a, const std::vector<float> &b, float expectedA, float expectedB, const std::string &testName) {
    bool testPassed = true;
    for (size_t i = 0; i < a.size(); ++i) {
        if (a[i] != expectedA || b[i] != expectedB) {
            testPassed = false;
            break;
        }
    }

    if (testPassed) {
        std::cout << testName << ": Passed" << std::endl;
    } else {
        std::cout << testName << ": Failed" << std::endl;
    }
}